GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -218,8 +219,37 @@ public virtual string DisplayText()
                 return ToString();
             }
 
+            protected void SetNodeFont(FontStyle style)
+            {
+                if (style == FontStyle.Regular)
+                {
+                    // For regular, set to null to use the NativeTreeView font
+                    if (TreeViewNode.NodeFont != null)
+                    {
+                        TreeViewNode.NodeFont.Dispose();
+                        TreeViewNode.NodeFont = null;
+                    }
+                }
+                else
+                {
+                    // If current font doesn't have the input style, get rid of it
+                    if (TreeViewNode.NodeFont != null && !TreeViewNode.NodeFont.Style.HasFlag(style))
+                    {
+                        TreeViewNode.NodeFont.Dispose();
+                        TreeViewNode.NodeFont = null;
+                    }
+
+                    // If non-null, our font is already valid, otherwise create a new one
+                    if (TreeViewNode.NodeFont == null)
+                    {
+                        TreeViewNode.NodeFont = new Font(AppSettings.Font, style);
+                    }
+                }
+            }
```

Comment:
Can we simplify it as follows?
```
            protected void SetNodeFont(FontStyle style)
            {
                if (TreeViewNode.NodeFont != null &&
                    (style == FontStyle.Regular || !TreeViewNode.NodeFont.Style.HasFlag(style)))
                {
                    TreeViewNode.NodeFont.Dispose();
                    TreeViewNode.NodeFont = null;
                }

                // If non-null, our font is already valid, otherwise create a new one
                if (TreeViewNode.NodeFont == null)
                {
                    TreeViewNode.NodeFont = new Font(AppSettings.Font, style);
                }
            }
```